File: 0057-insert-interval/0057-insert-interval.cpp
bool compare(vector<int>v1,vector<int>v2){
            return v1[0]<v2[0]?true:(v1[0]==v2[0]?(v1[1]<v2[1]?true:false):false);
}



class Solution {
public:
    vector<vector<int>> insert(vector<vector<int>>& intervals, vector<int>& newInterval) {
        
            int i=0;
            vector<vector<int>>ans;
            //ans.push_back(intervals[0]);
            if(intervals.size()==0){
                    return {newInterval};
            }
            bool flag=true;
        
            while(i<intervals.size()){
                       
                    if(ans.size()!=0 and intervals[i][0]<=ans.back()[1]){
                            if(intervals[i][1]<ans.back()[1]){
                                    i++;
                                    continue;
                            }
                            int temp=ans.back()[0];
                            ans.pop_back();
                            ans.push_back({temp,intervals[i][1]});          
                    }
                    else{
                                ans.push_back(intervals[i]);           
                    }
                    i++;
                        if(ans.back()[1]<newInterval[0] or ans.back()[0]>newInterval[1]){
                                
                                continue;
                        }

                        else if(ans.back()[1]>=newInterval[0] and ans.back()[1]<=newInterval[1]){

                                    flag=false;
                                    cout<<"here";
                                    if(newInterval[0]<=ans.back()[0]){

                                                ans.pop_back();
                                                ans.push_back(newInterval);

                                    }
                                    else{

                                            int temp = ans.back()[0];
                                            cout<<ans.size()<<endl;
                                            ans.pop_back();
                                            
                                            ans.push_back({temp,newInterval[1]});
                                    } 

                        }

                        else if(ans.back()[1]>=newInterval[1] and ans.back()[0]>=newInterval[0]){

                                    flag=false;
                                   cout<<"here";
                                   int temp=ans.back()[1];
                                   ans.pop_back();
                                   ans.push_back({newInterval[0],temp});

                        }  
                        else{
                            
                            flag=false;
                            
                        }
            }
        
            
            if(flag){
                    
                    ans.push_back(newInterval);
            }
        
           sort(ans.begin(),ans.end(),compare);
            return ans; 
    }
};

File: 0057-insert-interval/0057-insert-interval.py
class Solution:
    
    
    def insert(self, intervals: List[List[int]], newInterval: List[int]) -> List[List[int]]:
    
        if len(intervals)==0:
                return [newInterval]
        
        i=0
        ans=[]
        flag=False
        for item in intervals:
            
            
            if not flag:
                if item[1]<newInterval[0]:
                    
                        ans.append(item)
                else:
                        flag=True
                        if newInterval[1]<item[0]:
                                ans.append(newInterval)
                                ans.append(item)
                        else:
                                ans.append([min(item[0],newInterval[0]),max(item[1],newInterval[1])])
                                
            else:
                
                if item[0]>ans[-1][1]:
                        ans.append(item)
                        
                else:
                    
                    val=ans.pop(-1)
                    ans.append([val[0],max(val[1],item[1])])
        
        
        if not flag:
                ans.append(newInterval)
        
        return ans        